modules/images/webp-uploads/load.php

```diff
@@ -498,6 +512,20 @@ function webp_uploads_img_tag_update_mime_type( $image, $context, $attachment_id
 			$size_data['sources'][ $target_mime ]['file'],
 			$image
 		);
+
+		/**
+		 * Filter to replace additional image source file, by locating the original
+		 * mime types of the file and return correct file path in the end.
+		 *
+		 * @since n.e.xt
+		 *
+		 * @param string $image         An <img> tag where the urls would be updated.
+		 * @param int    $attachment_id The ID of the attachment being modified.
+		 * @param string $size          The size name that would be used to create this image, out of the registered subsizes.
+		 * @param string $target_mime   The target mime in which the image should be created.
+		 * @param string $context       The context where this is function is being used.
+		 */
+		$image = (string) apply_filters( 'webp_uploads_pre_replace_additional_image_source', $image, $attachment_id, 'full', $target_mime, $context );
```

Comment:
This filter should be placed above the line 510 and the code from `str_replace` would be executed only if the image is different, take a look at `suggested approach` here: 

- https://github.com/WordPress/performance/issues/311

The `full` value in this instance should be the name of the size being modified at this point in time.

Reply:
A couple of tests would be great as well.

Reply:
@mitogh I have added changes as per your feedback in the latest commits.